### ISSUE TITLE
Luminosity function in cosmoDC2

### ISSUE THREAD

**Issue by marina-ricci** [LSSTDESC/DC2-analysis#48]
Hi all 
 I am wondering how cosmoDC2 (v1.1.4) is constructed in terms of galaxy luminosity (not sure it is the right place to ask). The distribution of apparent magnitude vs redshift is weird: why is there no faint galaxies at low redshift?  The following plot shows only galaxies inside massive halos but the same is true for all galaxies. 
![image](https://user-images.githubusercontent.com/45492897/49960635-e4eb8500-ff10-11e8-9f21-0e975c083609.png)

When looking at the apparent luminosity function of galaxies in halos this feature appears as a sharp cut-off, whose position depends on redshift. 
![image](https://user-images.githubusercontent.com/45492897/49960571-b9689a80-ff10-11e8-87f9-99baf8782dca.png)

**Comment by yymao** [comment]
I'll invite @aphearin to answer this.

**Comment by aphearin** [comment]
Since there is a fixed cutoff in stellar mass (i.e., spectral normalization), shouldn’t this naturally give rise to increasing abundance of galaxies with fainter apparent magnitudes at higher redshift?

**Comment by marina-ricci** [comment]
Ok, I did not know that the catalog was constructed like this (where to find the info?). I find it very strange however, as it is not realistic in terms of magnitudes distribution. We are missing a lot of objects at low redshift, isn't it a problem for, e.g., image simulations ?

**Comment by aphearin** [comment]
@marina-ricci - the lack of low-z galaxies with faint apparent magnitudes is expected from any synthetic catalog with a physical resolution limit (such as a cutoff in physical mass, or a cutoff in restframe absolute magnitude, both of which follow directly from the simulation's finite mass resolution). My apologies if I'm misunderstanding your concern, so please help me: I don't follow why this diagram implies we are missing low-redshift galaxies. Could you be a bit more explicit about specifically which galaxies you think are missing, and why you think this is a problem?

**Comment by boutigny** [comment]
@aphearin I may miss something, but if the resolution limit leads to an unphysical lack of faint galaxies at low redshift, it means that our simulated images will also miss these galaxies. Extrapolating Marina's second plot (especially the blue curve corresponding to 0.25 < z < 0.35) we see that the missing galaxies, if present,  would fall in a magnitude range where they would contribute to the images.

**Comment by rmandelb** [comment]
@boutigny - I read @marina-ricci 's concern the same way you did.  For what it's worth, I agree it may technically be an issue at some low level, but not enough to result in a failure to meet the analysis working group's needs.  In particular:

- In the images, we want the number counts (dN/dmag) to be realistic.
- We also want the dN/dz for an i<25 sample to be realistic, i.e., reasonably consistent with an extrapolation of DEEP2 redshift distributions.

Since we know the dN/dmag is realistic, the results shown here imply that for galaxies at say 25<i<27, the redshift distribution is a bit off (i.e., we have the right number of them but if some are missing at low redshift, they must be shifted to high redshift).  However, the volume at low redshift is small and so it's likely a modest variation in the dN/dz.  A 1D slice through the 2D plot above, for galaxies of this magnitude range, would likely show this. It's important to keep in mind that these would appear in the images but generally not in a WL/LSS/CL-selected sample because they are sufficiently faint, so we are less concerned about their redshift and more just about their presence in the images.

In short, I agree with both of you that the effect is likely there according to this diagram, but it should not cause a significant problem for the science we're trying to do.

**Comment by aphearin** [comment]
Ahh, ok, now I get it, my mistake for being dense. @boutigny and @marina-ricci are not worried _per se_ about the generic trend, but rather you're worried about the cutoff at the faint end of the luminosity function at being too bright at low redshift. 

First, please have a look at my reproduction of the plot under discussion:

![faint_lf_lowz_check](https://user-images.githubusercontent.com/6951595/50019538-b0231080-ff98-11e8-85df-b476aaf3f733.png)

The black horizontal dashed line guides the eye to the m_i=26.5 Gold Sample limit. Looking at _all_ galaxies tells us that the apparent luminosity function only drops below 26.5 at z<~0.15, not as high as it appears to fall below when only looking at cluster galaxies. 

That being said, the criteria we received from DESC to calibrate the faint end of the luminosity function was the HSC dn/dmag test, [shown here](https://portal.nersc.gov/project/lsst/descqa/v2/?run=2018-11-28_5&test=ApparentMagFuncTest_HSCi&right=2018-11/2018-11-28_5/ApparentMagFuncTest_HSCi/summary.png), implemented primarily by @duncandc, with lots of guidance from @rmandelb. 

For reference, here is a plot showing the stellar mass range vs. redshift for these very faint galaxies. 

![faint_smf_lowz_check](https://user-images.githubusercontent.com/6951595/50019547-b9ac7880-ff98-11e8-9f3f-20a0ac8bd30c.png)

**Comment by boutigny** [comment]
Thank you very much @aphearin and @rmandelb for the very clear explanations

**Comment by rmandelb** [comment]
I think there is no further action needed on this issue, so am closing.  @marina-ricci if there are further concerns, please feel free to re-open for further discussion.
